src/input/input_string.rs

```diff
@@ -112,10 +112,7 @@ impl<'a> Input<'a> for StringMapping<'a> {
 
     fn validate_int(&'a self, _strict: bool) -> ValResult<ValidationMatch<EitherInt<'a>>> {
         match self {
-            Self::String(s) => match py_string_str(s)?.parse() {
-                Ok(i) => Ok(ValidationMatch::strict(EitherInt::I64(i))),
-                Err(_) => Err(ValError::new(ErrorTypeDefaults::IntParsing, self)),
-            },
+            Self::String(s) => str_as_int(self, py_string_str(s)?).map(ValidationMatch::strict),
```

Comment:
who is one `ValidationMatch::strict` and one `ValidationMatch::lax`?

Reply:
maybe add a test for both cases.

Reply:
Not sure, though I just kept the logic consistent with what was there before...

Reply:
This will control union behaviour, to add test cases would need to put these in unions with `str` as the equivalent validator on the RHS, and confirm that in both cases the `str` leg wins?

e.g. `dict[int, str] | dict[str, str]` - I guess the `dict[str, str]` is the better choice.

Reply:
As discussed with @sydney-runkle offline, I think the divergence is a historical bug that's probably on me. Let's not block this PR on it.